### DISCUSSION TITLE
Do we have to integrate with all Banks/ODFIs?

### DISCUSSION THREAD

**Discussion by GrahamMcBain** [moov-io/ach#969] [General]
Quick (naive) question: our product needs to support ACH across any of the banks of our customers and we don’t want to get into the escrow model (with a single partner bank and we do a pull+push). So that leaves us, I guess integrations with each of the ODFIs and we do the orchestration ourselves. The question is: what are the top ODFIs that have he largest coverage and if there are any specific requirements/roadblocks that we should anticipate when we integrate with each of them? Any stories/experiences on this is highly appreciated!

(pasted from the Moov slack channel)

**Comment by GrahamMcBain** [comment]
I think there may be some confusion on what your relationship with your ODFI will look like so I hope this clears things up! 

1. You only need a relationship with one bank in order to do pushes/pulls from every bank in the ACH network.
2. Your bank will most likely give you an FTP server through which you can upload your ACH files generated with this library. 
3. Within that file you can initiate pushes/pulls from any bank on the ACH network 

example

> Suzy needs to send $50 to Tom. Suzy is a customer of Bank A, Tom is a customer of Bank B and your ODFI is Bank C. You 
 create an ach file specifying the transfer from Suzy to Tom, and upload that file to Bank C's portal. When that file gets to the clearing house the instructions are sent to Bank A to debit $50 from Suzy's account and instructions are send to Bank B to credit Tom's account $50

I hope this makes sense/helps!

**Reply by ramkumarvenkat** [reply]
Thanks @GrahamMcBain - this means we have to become a TPSP with NACHA I guess (and not third-party sender because we are not an escrow)
